### PR TITLE
Deploy build artifact to all environments

### PR DESCRIPTION
**Pull request by enformatique11**
…napp-name

### CODE REVIEW

File: DqtNotifications/.gitignore [hunk 1]
```diff
@@ -399,4 +399,4 @@ FodyWeavers.xsd
 
 .idea
 
-src/DqtReporting.ConfigureSolution/Properties/launchSettings.json
\ No newline at end of file
+local.settings.json
```
Comment (gunndabad) [inline DqtNotifications/.gitignore:402]: We still need `src/DqtReporting.ConfigureSolution/Properties/launchSettings.json` here

File: .github/actions/deploy-environment/action.yml [hunk 2]
```diff
@@ -62,7 +62,23 @@ runs:
     - id: terraform
       run: |
         make ci ${{ inputs.environment_name }} terraform-apply
+         cd terraform
+        OUTPUTS=($(terraform output --json | jq -r 'keys | @sh' | tr -d \'))
+        for o in "${OUTPUTS[@]}"
+        do
+          echo "${o}=$(terraform output -raw ${o})" >> $GITHUB_ENV
```
Comment (gunndabad) [inline .github/actions/deploy-environment/action.yml:69]: Is this required?

Comment (enformatique11) [reply .github/actions/deploy-environment/action.yml:69]: Yes, it is required.
The reason the scripts produce the output from terraform:
 output "functionapp_name" {
  value = azurerm_linux_function_app.function_app.name
}
and this value is used here: app-name: ${{ env.functionapp_name }}
 uses: Azure/functions-action@v1
      with:
        app-name: ${{ env.functionapp_name }}
        package: ${{ github.workspace }}/artifact

TO USE FUNCTION APP NAME FOR DEV, TEST, PRE-PROD AND PRODUCTION ENVIRONMENTS.

File: DqtNotifications/.gitignore (hunk 1 above)
Comment (enformatique11) [reply DqtNotifications/.gitignore:402]: Fixed

**Comment by enformatique11** [comment]
[tra-dqt-notifications complete deployment workflow](https://trello.com/c/8NFgJzVF/450-tra-dqt-notifications-complete-deployment-workflow)

**Review by gunndabad — APPROVED** [review]

**Pull request merged**